pocs-capstone/backend/db/canvasrequests.py

```diff
@@ -0,0 +1,129 @@
+def canvas_request(url, headers, params):
+    response = requests.get(url=url, headers=headers, params=params) 
+    response.raise_for_status() # raise error if 4xx or 5xx
+    return response.json() #convert to json
+    
+'''Return a list of all Canvas courses' IDs'''
+def get_courses(canvas_token): #later we'll add userId as a parameter
+    auth_header = {'Authorization': 'Bearer ' + canvas_token}
+    courses_params = {
+    "per_page": 100,
+    "enrollment_state": "active",
+    "workflow_state": "available",
+    "enrollment_type": "student"
+    }
+
+    courses_data = canvas_request(BASE_URL + '/courses', auth_header,  courses_params)
+    course_id_list = [] # a list of all the user's courses (their ids)
+    for course_entry in courses_data:
+        course_id_list.append(course_entry['id']) # + ' ' + str(course_entry['name'])) # uncomment to print the name of the course too
+
+    return course_id_list
+
+
+'''Given a course ID, return a list of all assignments' IDs'''  
+def get_assignments(canvas_token, course_id): 
+    auth_header = {'Authorization': 'Bearer ' + canvas_token}
+    assignment_params = {
+        "per_page": 100,
+        "include": "submission"
+    }
+
+    assignments_data = canvas_request(BASE_URL + '/courses/' + str(course_id) + '/assignments', auth_header,  assignment_params)
+    assignment_id_list = [] # a list of all the user's courses (their ids)
+    for assignment_entry in assignments_data:
+        assignment_id_list.append(assignment_entry['id']) # + ' ' + str(assignment_entry['name'])) # uncomment to print the name of the course too
+
+    return assignment_id_list
+
+'''Given a course ID, return a json object of course information for that particular course'''  
+def get_course_info(canvas_token, course_id):
+    auth_header = {'Authorization': 'Bearer ' + canvas_token}
+    course_url = BASE_URL + '/courses/' + str(course_id)
+    return canvas_request(url=course_url, headers=auth_header, params={})
+
+'''Given a course ID and assignment ID, return a dict of assignment information for that particular assignment'''  
+def get_assignment_info(canvas_token, course_id, assignment_id):
+    auth_header = {'Authorization': 'Bearer ' + canvas_token}
+    assignment_url = BASE_URL + '/courses/' + str(course_id) + '/assignments/' + str(assignment_id)
+    a = canvas_request(url=assignment_url, headers=auth_header, params={"include[]":['submission']})
+    due = a['due_at']
+    if due != None:
+        due = due[0:10]
+    return {'title': a['name'] or "No title.",
+        'due_date': due,
+        #'created_date': a['created_at'],
+        #'completed_date': a['submission']['graded_at'],
+        #'completed': a['graded_submissions_exist'],
+        #'active': 'true',
+        'task_type': 'S',
+        #'task_level': 1,
+        #'recurring': 'false',
+        #'recurring_time_delta': 0,
+        'description': bs.BeautifulSoup(a['description'],'lxml').get_text() or "No description",
```

Comment:
Hello! I was getting an error because one of my assignments had 'None' returned for the description. The BeautifulSoup constructor was passed that 'None' object and I kept getting: `TypeError: object of type 'NoneType' has no len().`

I updated the line to look like : `bs.BeautifulSoup(a['description'],'lxml').get_text() if a['description'] != None else "No description",`. 

There are probably better ways of checking the value of a['description'] but this is what I have suggested for now!